### PR TITLE
Process oscpe rules events type

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#22140|

## Description 

This PR adds the needed changes to process oscpe events. 

## Logs/Alerts example

Ejecution with databasefeedmanager testtool 
image

Request rocksdb database content for oscpe_rules
image

Execution vulnerability scanner testtool for OSINFO type messages (input json added as file3.json and file4.json)
dbsync type message

rsync type message

### Refactor testing 

Filling vendor_map and oscpe_rules information 
image
image
image

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 1]
```diff
@@ -621,6 +621,20 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         return {};
     }
 
+    /**
+     * @brief Get OS CPE rules.
+     * @param osCpeRules returned information.
+     */
+    void getOsCpeRules(nlohmann::json& osCpeRules)
+    {
+        rocksdb::PinnableSlice ret;
```
Comment (GabrielEValenzuela) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:630]:
```suggestion
        rocksdb::PinnableSlice queryResult;
```

I think we can make a little improve in the variable name here

**Review by GabrielEValenzuela — COMMENTED** [review]
Amazing work @MiguelazoDS ! Just one little thing to adjust

- Code 🟢 
  - Complexity 🟡: 
    - In general, the CL is realy easy to follow and understand
    - There are some places where you could make things more clear
  - Design 🟢: The CL applies the use of already defined classes and coexists with other modules
  - Functionality ⚪ : 
    - TIP
- Testing ⚪ 
  - TIP

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp]: Changed

**Review by sebasfalcone — APPROVED** [review]
Amazin PR! 🥇 

- Code 🟢 
  - Complexity: Clean and easy to follow
  - Design: Reusability for new global maps! 🥇 
  - Functionality: Works as expected
 
- Test 🟢 

Adding a new vulnerability to a centOS 7 machine:
```
2024/02/29 20:40:27 wazuh-modulesd:vulnerability-scanner[361604] packageScanner.hpp:79 at operator()(): DEBUG: The platform is in the list based on CPE comparison for Package: libid3tag-devel, Version: 0.15.1b-17.el7, CVE: CVE-2017-11551, Content platform CPE: cpe:/o:redhat:enterprise_linux:7 OS CPE: cpe:/o:redhat:enterprise_linux:7
2024/02/29 20:40:27 wazuh-modulesd:vulnerability-scanner[361604] packageScanner.hpp:259 at operator()(): DEBUG: Match found for Package: libid3tag-devel for vulnerability: CVE-2017-11551 due to default status.
``` 

- It matches against its RedHat counterpart 🟢 
- It's indexed
image

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM!

**Pull request merged**